src/Tools/Performance Toolkit/App/src/BCPTTestSuite.Codeunit.al

```diff
@@ -411,4 +411,36 @@ codeunit 149006 "BCPT Test Suite"
         BCPTLine.SetRange("BCPT Code", SuiteCode);
         BCPTLine.SetRange("Codeunit ID", CodeunitID);
     end;
+
+    #region BCPTLineCodunitFacade
+    /// <summary>
+    /// Wrapper for OnBeforeBCPTLineAddLogEntry since it is not possible subscribe to an event in a object with scope internal
+    /// This one converts BCPTLine to variables instead of a record.
+    /// </summary>
+    /// <param name="BCPTLine"></param>
+    /// <param name="Operation"></param>
+    /// <param name="ExecutionSuccess"></param>
+    /// <param name="Message"></param>
+    /// <param name="Handled"></param>
+    internal procedure OnBeforeBCPTLineAddLogEntryWrapper(var BCPTLine: Record "BCPT Line"; var Operation: Text; var ExecutionSuccess: Boolean; var Message: Text; var Handled: Boolean)
```

Comment:
```suggestion
    internal procedure RaiseOnBeforeBCPTLineAddLogEntry(var BCPTLine: Record "BCPT Line"; var Operation: Text; var ExecutionSuccess: Boolean; var Message: Text; var Handled: Boolean)
```

Reply:
I don't understand the purpose of this method. Why would an outside caller need to raise the event directly and out of context?

Reply:
I'm not sure whether it's currently an acceptable solution, but I saw many events in the base app/system apps that are internal. If it's allowed, I believe it would be better than having a wrapper method.

Reply:
I think you are mixing up two things. No, you cannot subscribe to an event in an internal codeunit, but your event does not have to be on an internal codeunit. Almost none of our events are. The event publisher is placed on the public facade codeunit (which is where you have it too) as an internal method (and yes, you can subscribe to that). 
Now, you can call that method directly from the internal object and everyone can subscribe to it. You do not need any of the wrapper methods you created.